src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -210,4 +266,61 @@ class EntityClient (requestContext: RequestContext)(implicit protected val execu
       }
     }
   }
+
+  def importBagit(pipeline: WithTransformerConcatenation[HttpRequest, Future[HttpResponse]], workspaceNamespace: String, workspaceName: String, bagitRq: BagitImportRequest): Future[PerRequestMessage] = {
+    if(bagitRq.format != "TSV") {
+      Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, "Invalid format; for now, you must place the string \"TSV\" here"))
+    } else {
+
+      //Java URL handles http, https, ftp, file, and jar protocols.
+      //We're only OK with https to avoid MITM attacks.
+      val bagitURL = new URL(bagitRq.bagitURL)
+      val acceptableProtocols = Seq("https") //for when we inevitably change our mind and need to support others
+      if (!acceptableProtocols.contains(bagitURL.getProtocol)) {
+        Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, "Invalid bagitURL protocol: must be https only"))
+      } else {
+
+        val rand = java.util.UUID.randomUUID.toString.take(8)
+        val bagItFile = File.createTempFile(s"$rand-samples", ".tsv")
+
+        try {
+          val conn = bagitURL.openConnection()
+          val length = conn.getContentLength
+
+          if (length > 1000000) {
+            Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, s"BDBag size is too large."))
+          } else {
+            //this magic creates a process that downloads a URL to a file (which is #>), and then runs the process (which is !!)
+            bagitURL #> bagItFile !!
+
+            //make two big strings containing the participants and samples TSVs
+            //if i could turn back time this would use streams to save memory, but hopefully this will all go away when entity service comes along
+            unzipTSVs(bagitRq.bagitURL, new ZipFile(bagItFile.getAbsolutePath)) { (participantsStr, samplesStr) =>
+              (participantsStr, samplesStr) match {
+                case (None, None) =>
+                  Future.successful(RequestCompleteWithErrorReport(StatusCodes.BadRequest, "You must have either (or both) participants.tsv and samples.tsv in the zip file"))
+                case _ =>
+                  for {
+                    // This should toss back errors from rawls.
```

Comment:
minor, but the word "toss" has connotations in British English that make this comment extremely stick out for me. i will politely defer to [Stack Exchange](https://english.stackexchange.com/questions/409677/how-popular-is-the-slang-usage-of-toss-in-british-english) for a discussion of the vulgar details

Reply:
Alright, will change to "vomit"